src/targets/gpu/CMakeLists.txt

```diff
@@ -125,6 +219,41 @@ add_library(migraphx_gpu
 )
 set_target_properties(migraphx_gpu PROPERTIES EXPORT_NAME gpu)
 
+if(WIN32)
+    target_compile_options(migraphx_gpu
+        PRIVATE
+            -fdelayed-template-parsing
+            -fms-extensions
+            -fms-compatibility)
+
+    target_sources(migraphx_gpu
+        PRIVATE
+            kernels/migraphx_kernels.cpp
+            kernels/migraphx_kernels.hpp
+            kernels/resource.h
+            kernels/resource.rc)
+
+    target_include_directories(migraphx_gpu PRIVATE kernels)
```

Comment:
You are adding the resource files using `target_sources`. So instead of `target_sources(migraphx_gpu PRIVATE ...)` you can write `target_sources(${EMBED_NAME} INTERFACE ...)`. And when we do `target_link_libraries(migraphx_gpu PRIVATE kernels)` cmake will add the sources to `migraphx_gpu` as if we wrote `target_sources(migraphx_gpu PRIVATE ...)`.

Reply:
For now I added a simple fallback for the embed module in #1804. That should help get things to compile for windows and then we can look into a better design for this at a later time.

Reply:
Again, that is not how resource embedding works on Windows. The mechanism is __native__ to Windows, and toolchains have a dedicated resource compiler. The compiler produces a `*.res` binary file from the `*.rc` text file. Therefore, making it a separate static library or linking it explicitly to an executable or dynamic library is unnecessary. The linker automatically merges the `*.res` file to output binary. The only thing that needs to be done is adding a `*.rc` text file to the target's source files. Additionally, the WIN32 has a dedicated API to operate on resource file while embedded in a binary.

Reply:
Also, please do not push #1804 because that change will not help and will complicate porting even more (it is not portable).

Reply:
What you do in Embed.cmake is native to Linux (and only to Linux/UNIX). None other linkers provide such a functionality. Even llvm-lld (GNU ld emulation) on Windows cannot do that.

Reply:
> We will not use GNU for building on Windows because it introduces a non-native POSIX layer, and translations to and from WIN32 might impact the performance.

Yes of course, I wouldn't expect you to use GNU linker on windows.

In #1804 it will only use the GNU LD on linux. For other OSs, it will generate C arrays in source files so it will work portably for all OSs.  Later, we can work on building a windows-specific implementation for this in the future if we think that will be better than the generic portable version.